openreview/arr/process/configuration_process.py

```diff
@@ -0,0 +1,822 @@
+                'invitees': [openreview.stages.CustomStage.Participants.AUTHORS],
+                'readers': [],
+                'content': arr_author_consent_content,
+                'notify_readers': False,
+                'email_sacs': False
+            },
+            due_date=note.content.get('author_consent_due_date'),
+            exp_date=note.content.get('form_expiration_date')   
+        ),
+        ARRStage(
+            type=ARRStage.Type.CUSTOM_STAGE,
+            required_fields=['reviewer_checklist_due_date', 'reviewer_checklist_exp_date'],
+            super_invitation_id=f"{venue_id}/-/Reviewer_Checklist",
+            stage_arguments={
+                'name': 'Reviewer_Checklist',
+                'reply_to': openreview.stages.CustomStage.ReplyTo.FORUM,
+                'source': openreview.stages.CustomStage.Source.ALL_SUBMISSIONS,
+                'invitees': [openreview.stages.CustomStage.Participants.REVIEWERS_ASSIGNED],
+                'readers': [
+                    openreview.stages.CustomStage.Participants.SENIOR_AREA_CHAIRS_ASSIGNED,
+                    openreview.stages.CustomStage.Participants.AREA_CHAIRS_ASSIGNED,
+                    openreview.stages.CustomStage.Participants.REVIEWERS_ASSIGNED
+                ],
+                'content': arr_reviewer_checklist,
+                'notify_readers': False,
+                'email_sacs': False
+            },
+            due_date=note.content.get('reviewer_checklist_due_date'),
+            exp_date=note.content.get('reviewer_checklist_exp_date'),
+            process='process/checklist_process.py',
+            preprocess='process/checklist_preprocess.py',
+            extend=extend_reviewer_checklist
+        ),
+        ARRStage(
+            type=ARRStage.Type.CUSTOM_STAGE,
+            required_fields=['ae_checklist_due_date', 'ae_checklist_exp_date'],
+            super_invitation_id=f"{venue_id}/-/Action_Editor_Checklist",
+            stage_arguments={
+                'name': 'Action_Editor_Checklist',
+                'reply_to': openreview.stages.CustomStage.ReplyTo.FORUM,
+                'source': openreview.stages.CustomStage.Source.ALL_SUBMISSIONS,
+                'invitees': [openreview.stages.CustomStage.Participants.AREA_CHAIRS_ASSIGNED],
+                'readers': [
+                    openreview.stages.CustomStage.Participants.SENIOR_AREA_CHAIRS_ASSIGNED,
+                    openreview.stages.CustomStage.Participants.AREA_CHAIRS_ASSIGNED
+                ],
+                'content': arr_ae_checklist,
+                'notify_readers': False,
+                'email_sacs': False
+            },
+            due_date=note.content.get('ae_checklist_due_date'),
+            exp_date=note.content.get('ae_checklist_exp_date'),
+            process='process/checklist_process.py',
+            preprocess='process/checklist_preprocess.py',
+            extend=extend_ae_checklist
```

Comment:
what is this extend for?

Reply:
The `extend` functions are called as a post-processing step after the invitation is created if something about the invitation needs to be changed that wasn't supported by the built-in stages. The extend checklist functions modify the super invitations so that they are compatible with the ethics flagging code for changing the readers

The same idea for the `build_edit` functions because sometimes edits need to be built with some additional logic (though for the pre-print case I think it can be removed, it seems simple enough to put everything in the edit without additional code)